src/v/storage/segment_index.h

```diff
@@ -85,11 +99,14 @@ class segment_index {
 
 private:
     ss::sstring _name;
-    ss::file _out;
     size_t _step;
     size_t _acc{0};
     bool _needs_persistence{false};
     index_state _state;
+    debug_sanitize_files _sanitize;
+
+    // For unit testing, a hook to replace actually opening a file
```

Comment:
Can this comment be improved? In itself it is not a hook, just a `std::optional`.

E.g., can you add something like "if this is set, functions x,y do z instead of <their normal behavior>".

Reply:
Changed this to describe the behavioural change to open()